mlrun/runtimes/pod.py

```diff
@@ -619,6 +684,100 @@ def _prune_node_selector_requirements_from_node_selector_terms(
                 )
         return new_node_selector_terms
 
+    @staticmethod
+    def _get_current_project_name(owner: typing.Any) -> typing.Optional[str]:
+        metadata_object = getattr(owner, "metadata", None)
+        if metadata_object and getattr(metadata_object, "project", None):
+            return metadata_object.project
+        spec_object = getattr(owner, "spec", None)
+        if spec_object:
+            nested_metadata = getattr(spec_object, "metadata", None)
+            if nested_metadata and getattr(nested_metadata, "project", None):
+                return nested_metadata.project
+        return None
+
+    @classmethod
+    def _assert_secret_allowed(
+        cls, owner: typing.Any, secret_name: str, context: str
+    ) -> None:
+        if not secret_name:
+            return
+        if (
+            cls._FORBIDDEN_AUTH_SECRET_PATTERN.match(secret_name)
+            or secret_name == cls._SMTP_CONFIG_SECRET
+        ):
+            raise mlrun.errors.MLRunInvalidArgumentError(
+                f"Blocking secret '{secret_name}' in {context}: mounting auth-secret is not allowed on the client side"
+            )
+        project_match = cls._PROJECT_SECRET_PATTERN.match(secret_name)
+        if project_match:
+            target_project_name = project_match.group("project")
+            current_project_name = cls._get_current_project_name(owner)
+            if not current_project_name or target_project_name != current_project_name:
+                raise mlrun.errors.MLRunInvalidArgumentError(
+                    f"Blocking secret '{secret_name}' in {context}: project-secret for '{target_project_name}' "
+                    f"is not allowed for project '{current_project_name or 'unknown'}'"
+                )
+
+    @staticmethod
+    def _get_secret_name_from_env_value_from(
+        value_from: typing.Any,
+    ) -> typing.Optional[str]:
+        if value_from is None:
+            return None
+        # k8s object style
+        secret_key_ref_object = getattr(value_from, "secret_key_ref", None)
+        if secret_key_ref_object and getattr(secret_key_ref_object, "name", None):
+            return secret_key_ref_object.name
+        # dict style
+        if isinstance(value_from, dict):
+            possible_ref = (
+                value_from.get("secretKeyRef")
+                or value_from.get("secret_key_ref")
+                or value_from.get("valueFrom", {}).get("secretKeyRef")
+                or value_from.get("valueFrom", {}).get("secret_key_ref")
+            )
+            if isinstance(possible_ref, dict):
+                return possible_ref.get("name")
+        return None
+
+    @staticmethod
+    def _get_secret_name_from_volume(
```

Comment:
all these volumes/envs and other checks are already implemented in https://github.com/mlrun/mlrun/blob/7b1424a9f14da2ef7f9ec56cab40d3d7147a9825/server/py/framework/api/utils.py#L369-L371 and may be reused if passing validating function to them or just returning the list first and validating later.